### PR TITLE
Add validation for plate wells

### PR DESCRIPTION
**Pull request by elevu**
This PR limits the possibility of submitting samples in 96 well plates when the layout is not column ordered
I opened a new pull request since I rewrote the validation. It is too complex to be at the form level, especially for big amount of samples. 
Too make it more clear for the user the plates are now validated when clicking on submit sample.

**How to prepare for test**:
- [x] ssh to clinical-db
- [x] install on stage of clinical-db:
`bash servers/resources/clinical-db.sclifelab.se/update-cgweb-stage.sh [THIS-BRANCH-NAME]`

**How to test**:
- [x] login on https://clinical-stage.scilifelab.se/
- [ ] Upload different kind of orders with sample that well or not so well ordered.

**Expected test outcome**:
- [x] check that the validation works as expected
- [x] Take a screenshot and attach or copy/paste the output.

I tested the files uploaded by Anna https://drive.google.com/drive/u/0/folders/11JwqKgdaXoZE9X_A8uoVzF7YLj0Q7DGG
and updated the result here https://docs.google.com/spreadsheets/d/1iRflJ6JdIlkhoQhy0MevTFbJVQIs5wE_3WABKkisjzk/edit#gid=0

Now there is a clear message that appears describing what is wrong with the plate validation, for example:
Screen Shot 2020-11-04 at 13 05 37
Screen Shot 2020-11-04 at 12 56 30
Screen Shot 2020-11-04 at 12 54 18

**Review:**
- [ ] code approved by
- [x] tests executed by EM
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by elevu** [comment]
The branch is deployed on stage but the login is not working, I am getting {"message":"outdated login certificate"} from the backend. Maybe cg needs to be restarted @patrikgrenfeldt ?

**Comment by elevu** [comment]
now staging works!

File: utils/constants.js [hunk 1]
```diff
@@ -23,6 +23,105 @@ export const containerOptions = [
   "Tube",
 ];
 
+export const wellsLayout = {
```
Comment (patrikgrenfeldt) [inline utils/constants.js:26]: I think this layout is simple and understandable 👍

**Review by patrikgrenfeldt — CHANGES_REQUESTED** [review]
I think you have solved a complicated problem and I think we should discuss the solution when you have time to see if we can simplify it somehow?

**Comment by cesve** [comment]
@elevu perhaps Anna Lyander should review this PR?

**Review by patrikgrenfeldt — APPROVED** [review]
Massive feat 👍  

I think the plate[0] and plate[1] is a bit hard for me to understand what they are.
I can't judge if the the comparison to the length of the wells on line 364-365 is correct I wish it was written in a way I could intuitively read. Maybe get rid of the magic numbers and replace with abstractions like is_x_and_y methods..?

**Comment by elevu** [comment]
> @elevu perhaps Anna Lyander should review this PR?

@annaengstrom let me know if you prefer to test this again after I changed the validation or if I can merge.

**Comment by annaengstrom** [comment]
Expected result FAIL but PASS for some tests, see [here](https://docs.google.com/spreadsheets/d/1iRflJ6JdIlkhoQhy0MevTFbJVQIs5wE_3WABKkisjzk/edit#gid=0). Orderforms [here](https://drive.google.com/drive/u/0/folders/11JwqKgdaXoZE9X_A8uoVzF7YLj0Q7DGG).

Example, *19 samples* test file:

Screenshot 2020-11-12 at 12 12 51

**Comment by annaengstrom** [comment]
Random gaps in micro order also PASS, excepted to FAIL.

[MWL_more than88-random gaps_ 2020-10-29-1603-9_1603-9_Order_form_Microbial_WGS.xlsx](https://github.com/Clinical-Genomics/cgweb/files/5530213/MWL_more.than88-random.gaps_.2020-10-29-1603-9_1603-9_Order_form_Microbial_WGS.xlsx)

Screenshot 2020-11-12 at 12 17 30

**Comment by elevu** [comment]
You need to submit the sample to see the validation, you can see in the PR intro how the validation works now. It is moved to the "Submit" stage since it is too messy and resource consuming to validate hundreds of sample every time one of them is changed.

I get failed from _19 samples_ test file:

Screen Shot 2020-11-12 at 12 18 46

**Comment by annaengstrom** [comment]
@elevu I did the tests again, all test results are correct. Well done!

**Comment by elevu** [comment]
> @elevu I did the tests again, all test results are correct. Well done!

yayy! :raised_hands: 
Merging.

**Pull request merged**